### DISCUSSION TITLE
Is  __splink__df_concat_with_tf working correctly in Splink 3.0?

### DISCUSSION THREAD

**Discussion by MirandaLochner** [moj-analytical-services/splink#647] [General]
Getting an error message stating: "cannot resolve 'l.tf_FirstName' given input columns".

**Comment by RobinL** [comment]
Please could you post the code that's causing this error and the full stack trace so we can look into this.

If possible, could you also check whether lower casing all columns in the input dataframe solves this.

Thanks!

**Reply by MirandaLochner** [reply]
Yep! Also tried all lower case.
Error:
AnalysisException: cannot resolve 'l.tf_mailcity' given input columns:
l.tf_MailPostalCode, r.tf_MailPostalCode, l.unique_id, r.unique_id]

Settings:
        cl.levenshtein_at_thresholds("firstname",2),
        cl.exact_match("lastname", term_frequency_adjustments=True),
        cl.exact_match("dateofbirth"),
        cl.exact_match("mailcity", term_frequency_adjustments=True),

**Reply by MirandaLochner** [reply]
Actually I forgot to lower case my blocking keys. Lower case fixed the issue, thank you!

**Reply by RobinL** [reply]
Thanks.  We did try to solve some of these case issues already, but looks like we haven't been completely successful, so looks like we need to have another look
